tests/e2e/configurer/upgrade.go

```diff
@@ -147,10 +147,30 @@ func (uc *UpgradeConfigurer) RunSetup() error {
 }
 
 func (uc *UpgradeConfigurer) RunUpgrade() error {
+	var err error
 	if uc.forkHeight > 0 {
-		return uc.runForkUpgrade()
+		err = uc.runForkUpgrade()
+	} else {
+		err = uc.runProposalUpgrade()
 	}
-	return uc.runProposalUpgrade()
+	if err != nil {
+		return err
+	}
+
+	// Check if the nodes are running
+	for chainIndex, chainConfig := range uc.chainConfigs {
+		chain := uc.baseConfigurer.GetChainConfig(chainIndex)
+		for validatorIdx := range chainConfig.NodeConfigs {
+			node := chain.NodeConfigs[validatorIdx]
+			// Check node status
+			err = node.NodeStatus()
+			if err != nil {
+				uc.t.Logf("Node is not running, chain-id %s, node %s", chainConfig.Id, node.Name)
+				return err
+			}
+		}
```

Comment:
If there is no error, can we add a log that a node with id X and name Y has upgraded successfully please

Reply:
^^ Also following up on this, is it possible to add "good things to know" in e2e readme where you show this failure happening and ways to fix it?

Reply:
@hieuvubk can you please address this suggestion from stackman27 before we merge?

Reply:
yes sure I just missed this cmt thanks u @stackman27